ogorek.go

```diff
@@ -123,10 +124,37 @@ type Decoder struct {
 	line  []byte
 }
 
+// DecoderConfig allows to tune Decoder.
+type DecoderConfig struct {
+	// PersistentLoad, if !nil, will be used by decoder to handle persistent references.
+	//
+	// Whenever decoder finds one in pickle stream, it will call
+	// PersistentLoad with reference's persistent ID. If PersistentLoad
+	// returns !nil object without error, decoder will use that object
+	// instead of Ref in the resulted built Go object.
+	//
+	// An example use-case for PersistentLoad is to transform persistent
+	// references in a ZODB database of form (type, oid) tuple, into
+	// equivalent-to-type Go ghost object, e.g. equivalent to zodb.BTree.
+	//
+	// See Ref documentation for details and further pointers on the subject.
+	PersistentLoad func(pid interface{}) (interface{}, error)
```

Comment:
And then have this accept a `Ref` instead of an `interface{}` ?

Reply:
Thanks for feedback. I've actually started with PersistentLoad accepting Ref but than switched for it to accept Ref.Pid diectly, probably to mimic python api 1-1. On a second glance I agree that having less `interface{}` in the API would make things a bit more clear. Please let me know you preference - it should be easy for me to rework the patch to be

```
PersistentID func(obj interface{}) *Ref
PersistentLoad func(ref *Ref) (interface{}, error)
```